lib/engine/game/base.rb

```diff
@@ -477,11 +477,21 @@ def init_company_abilities
 
           case (share = ability[:share].to_s)
           when 'random-president'
-            share = @corporations[rand % @corporations.size].shares[0]
+            corporation = @corporations[rand % @corporations.size]
+            share = corporation.shares[0]
             ability[:share] = share
-            corporation = share.corporation
             company.desc = "#{company.desc} The random corporation in this game is #{corporation.name}."
             @log << "#{company.name} comes with the president's share of #{corporation.name}"
+          when 'random'
+            exclusions = ability[:'random-share-exclude']&.split
```

Comment:
i think it's easier to do inclusions as opposed to exclusions, thoughts?

inclusion also follows the pattern of other privates we have like what tiles and hexes to lay

Reply:
Agree on inclusions. That also how I rewrote it in the rules text to ease setup and understanding!